src/main/groovy/org/parchmentmc/compass/CompassExtension.groovy

```diff
@@ -0,0 +1,53 @@
+package org.parchmentmc.compass
+
+import groovy.transform.CompileStatic
+import net.minecraftforge.gdi.annotations.DSLProperty
+import net.minecraftforge.gdi.annotations.ProjectGetter
+import org.gradle.api.Project
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.provider.Property
+import org.parchmentmc.compass.storage.io.MappingIOFormat
+
+import javax.inject.Inject
+
+@CompileStatic
+abstract class CompassExtension {
+    private final Project project
+
+    @Inject
+    CompassExtension(final Project project) {
+        this.project = project
+        getLauncherManifestURL().convention('https://piston-meta.mojang.com/mc/game/version_manifest_v2.json');
+        getProductionData().convention(project.layout.projectDirectory.dir('data'))
+        getProductionDataFormat().convention(MappingIOFormat.ENIGMA_EXPLODED)
+        getStagingData().convention(project.layout.projectDirectory.dir('staging'))
+        getStagingDataFormat().convention(MappingIOFormat.ENIGMA_EXPLODED)
+        getInputs().convention(project.layout.projectDirectory.dir('input'))
```

Comment:
I believe these can be modified to access the properties as if they were fields?